include/uk/arch/spinlock.h

```diff
@@ -21,20 +23,62 @@
  * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
  * SUCH DAMAGE.
  */
-/* Taken from Mini-OS */
 
 #ifndef __UKARCH_SPINLOCK_H__
 #define __UKARCH_SPINLOCK_H__
 
+#include <uk/config.h>
+#include <stdint.h>
+
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-typedef struct {} spinlock_t;
+#ifdef CONFIG_UKPLAT_LCPU_MULTICORE
+/** spinlock struct */
+typedef struct {
+	volatile uint32_t lock;
+} __spinlock_t;
+
+/** section attribute cannot be used with pointers */
+typedef __spinlock_t *spinlock_ptr_t;
+
+/* spinlock type for declaration */
+#define spinlock_t __attribute__((section(".slocks"))) __spinlock_t
```

Comment:
We prefer to keep spinlocks in the regular data sections, so please remove this define and also the changes to the `sections.h` and `link64.lds.S`.
```suggestion

```